src/web/crate_details.rs

```diff
@@ -235,21 +236,11 @@ impl CrateDetails {
             crate_details.owners.push((row.get(0), row.get(1)));
         }
 
-        // retrieve last successful build if build failed
         if !crate_details.build_status {
-            let rows = conn.query(
-                "SELECT version
-                    FROM releases
-                    INNER JOIN crates ON releases.crate_id = crates.id
-                    WHERE build_status = true AND yanked = false AND crates.name = $1
-                    ORDER BY release_time desc
-                    LIMIT 1;",
-                &[&name],
-            ).unwrap();
-
-            if rows.len() >= 1 {
-                crate_details.last_successful_build = Some(rows.get(0).get(0));
-            }
+            crate_details.last_successful_build = crate_details.releases.iter()
+                .filter(|release| release.build_status && !release.yanked)
+                .map(|release| release.version.to_owned())
+                .nth(0);
```

Comment:
```suggestion
                .next();
```

Reply:
Aha, Rust always has a slighter better way of doing things.

Reply:
This is actually the way iterators are implemented under the hood :) If you look at https://doc.rust-lang.org/std/iter/trait.Iterator.html, `next()` is the only required function and all the rest are provided methods.

Reply:
Ah, that's right! I even knew that, but I guess I didn't connect the dots at the time.